File: TransCoder_tokenized_test_set_functions/python/N_TH_ROOT_NUMBER.py
def nthRoot ( A, N ) :
    xPre = random . randint ( 1, 101 ) % 10
    eps = 0.001
    delX = 2147483647
    xK = 0.0
    while ( delX > eps ) :
        xK = ( ( N - 1.0 ) * xPre + A / pow ( xPre, N - 1 ) ) / N
        delX = abs ( xK - xPre )
        xPre = xK;
        
    return xK

File: TransCoder_tokenized_test_set_functions/java/N_TH_ROOT_NUMBER.java
static double nthRoot ( int A, int N ) {
  double xPre = Math . random ( ) % 10;
  double eps = 0.001;
  double delX = 2147483647;
  double xK = 0.0;
  while ( delX > eps ) {
    xK = ( ( N - 1.0 ) * xPre + ( double ) A / Math . pow ( xPre, N - 1 ) ) / ( double ) N;
    delX = Math . abs ( xK - xPre );
    xPre = xK;
  }
  return xK;
}